Drivers/Periphs/Src/IMU.c

```diff
@@ -24,8 +26,27 @@
  * @return pdTRUE if initialization was successful, pdFALSE otherwise
  */
 BaseType_t IMU_Init(void) {
-  // TODO: Initialize SPI
-  // TODO: Initialize Sensor
+  int pTxData = 0;
+  int pRxData = 0;
+
+  IMUCommands_t IMUcommands;
+  //Initialize SPI
+  MX_SPI1_Init();
+
+  // Initialize IMU, gyroscope, and accelerometer
+  // NOTE: accelerometer and gyroscope features may be further configured, see IMU.h for more details
+  // TODO: Configure gyroscope and accelerometer
+  for(int i = 0; i < imuInit[0]; i ++){
+    pTxData = imuInit[i+1];
+    HAL_SPI_Transmit(&hspi1, &pTxData, 1, NO_POLLING);
+  }
+
+  // Initialize IMU hardware FIFO
+  // NOTE: FIFO size, rate and mode may be further configured, see IMU.h for more details
+  for(int i = 0; i < fifoInit[0], i ++){
+    pTxData = fifoInit[i+1];
```

Comment:
since you're iterating through a const array, there's no need to encode length, just use `sizeof(arr) / sizeof(*arr)`